Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIScriptingCocoa.mm

```diff
@@ -342,6 +349,31 @@ static inline void parseCSSInjectionOptions(NSDictionary *cssInfo, WebExtensionS
     return true;
 }
 
+NSArray* WebExtensionAPIScripting::toWebAPI(Vector<WebExtensionScriptInjectionResultParameters>& parametersVector)
+{
+    auto *results = [NSMutableArray arrayWithCapacity:parametersVector.size()];
+
+    for (auto& parameters : parametersVector) {
+        NSMutableDictionary *result = [NSMutableDictionary dictionaryWithCapacity:3];
+
+        if (parameters.result)
+            result[@"result"] = parameters.result.value();
+        else
+            result[@"result"] = NSNull.null;
+
+        ASSERT(parameters.frameID);
+        if (parameters.frameID)
+            result[@"frameId"] = @(WebKit::toWebAPI(parameters.frameID.value()));
+
+        if (parameters.error)
+            result[@"error"] = parameters.error.value();
```

Comment:
Ditto on the cast?

Reply:
I thought you could go from String to NSString * without a cast? but i'll add